lib/puma/dsl.rb

```diff
@@ -387,16 +393,26 @@ def threads(min, max)
     #   }
     def ssl_bind(host, port, opts)
       verify = opts.fetch(:verify_mode, 'none').to_s
-      no_tlsv1 = opts.fetch(:no_tlsv1, 'false')
-      no_tlsv1_1 = opts.fetch(:no_tlsv1_1, 'false')
+
+      tls_str = nil
+      tls_str = if opts[:no_tlsv1_1]
+        '&no_tlsv1_1=true'
+      elsif opts[:no_tlsv1]
+        '&no_tlsv1=true'
+      end
+
+      if opts[:no_tlsv1_3]
+        tls_str = "#{tls_str || ''}&no_tlsv1_3=true"
+      end
+
       ca_additions = "&ca=#{opts[:ca]}" if ['peer', 'force_peer'].include?(verify)
 
       if defined?(JRUBY_VERSION)
         keystore_additions = "keystore=#{opts[:keystore]}&keystore-pass=#{opts[:keystore_pass]}"
-        bind "ssl://#{host}:#{port}?cert=#{opts[:cert]}&key=#{opts[:key]}&#{keystore_additions}&verify_mode=#{verify}&no_tlsv1=#{no_tlsv1}&no_tlsv1_1=#{no_tlsv1_1}#{ca_additions}"
+        bind "ssl://#{host}:#{port}?cert=#{opts[:cert]}&key=#{opts[:key]}&#{keystore_additions}&verify_mode=#{verify}#{tls_str}#{ca_additions}"
```

Comment:
I've just realized how silly it is that we pass around arguments to `bind` internally as a string rather than as arguments. File that one away for the future...